core/thread.py

```diff
@@ -859,11 +859,25 @@ def create(
         self.cache[recipient.id] = thread
 
         # Schedule thread setup for later
+        cat = self.bot.main_category
+        if len(cat.channels) == 50:
+            fallback_id = self.bot.config["fallback_category_id"]
+            fallback = discord.utils.get(cat.guild.categories, id=int(fallback_id))
+            if fallback and len(fallback.channels) != 50:
+                self.bot.loop.create_task(thread.setup(creator=creator, category=fallback))
+                return thread
+
+            fallback = await cat.clone(name="Fallback Modmail")
+            self.bot.config.set("fallback_category_id", fallback.id)
+            await self.bot.config.update()
+            self.bot.loop.create_task(thread.setup(creator=creator, category=fallback))
+            return thread
+
         self.bot.loop.create_task(thread.setup(creator=creator, category=category))
         return thread
```

Comment:
Instead of keeping three references for creating the thread (line 687, 874, 876), use `if`s and redefine `cat` to the new category's ID, and keep only line 876 for creating the new thread.

Reply:
Great Idea! done.